src/main/kotlin/no/nav/helsearbeidsgiver/soeknad/SoeknadRepository.kt

```diff
@@ -47,4 +49,65 @@ class SoeknadRepository(
                 .map { it[sykepengesoeknad] }
                 .firstOrNull()
         }
+
+    fun hentSoeknaderMedVedtaksperiodeId(vedtaksperiodeId: UUID): List<SykepengesoknadDTO> =
+        transaction(db) {
+            SoeknadEntitet
+                .selectAll()
+                .where {
+                    SoeknadEntitet.vedtaksperiodeId eq vedtaksperiodeId
+                }.map { it[sykepengesoeknad] }
+        }
+
+    fun oppdaterSoeknaderMedVedtaksperiodeId(
+        soeknadIder: Set<UUID>,
+        vedtaksperiodeId: UUID,
+    ) {
+        try {
+            transaction(db) {
+                val eksisterendeSoeknader =
+                    SoeknadEntitet
+                        .selectAll()
+                        .where {
+                            soeknadId inList soeknadIder
+                        }.associate { it[soeknadId] to it[SoeknadEntitet.vedtaksperiodeId] }
+                loggDuplikateOgManglendeSoeknader(eksisterendeSoeknader, vedtaksperiodeId, soeknadIder)
+                val resterendeSoeknader = eksisterendeSoeknader.filter { it.value == null }.keys
+                if (resterendeSoeknader.isNotEmpty()) {
+                    SoeknadEntitet
+                        .updateReturning(
+                            returning = listOf(soeknadId),
+                            where = { soeknadId inList resterendeSoeknader },
+                        ) {
+                            it[SoeknadEntitet.vedtaksperiodeId] = vedtaksperiodeId
+                        }.map { it[soeknadId] }
+                        .also { logger().info("Oppdaterte søknader $it med vedtaksperiodeId: $vedtaksperiodeId") }
```

Comment:
Er det nødvendig å bruke returning her? Virker som vi har alle verdiene vi trenger for å logge uten å bruke returning.